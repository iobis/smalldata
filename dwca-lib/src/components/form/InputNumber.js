import classNames from 'classnames'
import i18next from 'i18next'
import PropTypes from 'prop-types'
import React from 'react'
import { useTranslation } from 'react-i18next'

export default function InputNumber({ className, name, onChange, optional, value, step }) {
  const { t } = useTranslation()
  const label = t(name + '.label')
  const placeholderKey = name + '.placeholder'
  const helpKey = name + '.help'

  return (
    <div className={classNames('column field', className)}>
      <label className={classNames('label', { 'has-text-weight-normal': optional })}>
        {label}
      </label>
      <input
        className="input"
        onChange={(e) => onChange(parseFloat(e.target.value))}
        placeholder={i18next.exists(placeholderKey) ? t(placeholderKey) : undefined}
        step={step}
        type="number"
        value={value === undefined || value === null ? '' : value}/>
      {i18next.exists(helpKey) ? <p className="help">{t(helpKey)}</p> : null}
    </div>
  )
}

InputNumber.propTypes = {
  className: PropTypes.string,
  name:      PropTypes.string.isRequired,
  onChange:  PropTypes.func.isRequired,
  optional:  PropTypes.bool,
  step:      PropTypes.number,
  value:     PropTypes.number
}
