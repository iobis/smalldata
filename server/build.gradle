plugins {
  id 'java'
  id 'application'
  id "io.franzbecker.gradle-lombok" version "2.1"
  id 'com.github.johnrengelman.shadow' version '4.0.4'
  id 'org.javamodularity.moduleplugin' version '1.4.0'
  id 'org.beryx.runtime' version '1.1.5'
  id 'checkstyle'
  id 'pmd'
}

repositories {
  mavenLocal()
  jcenter()
}

ext {
  vertxVersion = '3.6.3'
  jacksonVersion = '2.9.8'
  junitJupiterEngineVersion = '5.2.0'
  moduleName = 'obis.smalldata'
}

sourceCompatibility = 10

group = 'org.obis'
version = '1.0.0-SNAPSHOT'

mainClassName = "io.vertx.core.Launcher"

def mainVerticleName = "org.obis.smalldata.Starter"
def watchForChange = 'src/**/*'
def doOnChange = './gradlew classes'

runtime {
  options = ['--strip-debug', '--compress', '2', '--no-header-files', '--no-man-pages']
}

dependencies {
  implementation "io.vertx:vertx-core:$vertxVersion"
  implementation "io.vertx:vertx-web:$vertxVersion"
  implementation "io.vertx:vertx-web-client:$vertxVersion"
  implementation "io.vertx:vertx-mongo-client:$vertxVersion"
  implementation "io.vertx:vertx-web-api-contract:$vertxVersion"

  implementation "com.fasterxml.jackson.dataformat:jackson-dataformat-xml:$jacksonVersion"
  implementation "com.fasterxml.jackson.datatype:jackson-datatype-jsr310:$jacksonVersion"

  implementation "de.flapdoodle.embed:de.flapdoodle.embed.mongo:2.2.0"
  implementation "org.tinylog:tinylog:1.3.5"
  implementation "org.tinylog:slf4j-binding:1.3.5"

  compileOnly "org.projectlombok:lombok:1.18.6"

  testImplementation "io.vertx:vertx-junit5:$vertxVersion"
  testImplementation "io.vertx:vertx-junit5:$vertxVersion"
  testImplementation "com.fasterxml.uuid:java-uuid-generator:3.1.5"
  testImplementation "commons-io:commons-io:2.6"

  testRuntime("org.junit.jupiter:junit-jupiter-engine:$junitJupiterEngineVersion")
}

test {
  useJUnitPlatform()
  testLogging {
    events 'PASSED', 'FAILED', 'SKIPPED'
  }
}

run {
  args = ['run', mainVerticleName, "--redeploy=$watchForChange", "--launcher-class=$mainClassName", "--on-redeploy=$doOnChange"]
}

shadowJar {
  manifest {
    attributes 'Main-Verticle': mainVerticleName
  }
  mergeServiceFiles {
    include 'META-INF/services/io.vertx.core.spi.VerticleFactory'
  }
}

checkstyle {
  ignoreFailures false
  maxErrors 0
  maxWarnings 0
  toolVersion "8.17"
}

pmd {
  consoleOutput true
  ruleSetConfig = resources.text.fromFile("config/pmd/ruleset.xml")
  ruleSets = []
}
