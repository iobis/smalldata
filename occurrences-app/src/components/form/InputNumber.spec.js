import InputNumber from './InputNumber'
import React from 'react'
import { mount } from 'enzyme/build'

describe('InputText', () => {
  it('renders correctly', () => {
    expect(mount(createComponent())).toMatchSnapshot()
  })

  it('renders correctly if optional', () => {
    expect(mount(createComponent({ optional: true }))).toMatchSnapshot()
  })

  it('returns integer number when changing value to integer', () => {
    const onChange = jest.fn()
    const wrapper = mount(createComponent({ onChange }))

    wrapper.find('input').simulate('change', { target: { value: '1234' } })
    expect(onChange).toHaveBeenCalledTimes(1)
    expect(onChange).toBeCalledWith(1234)
  })

  it('returns float number when changing value to float', () => {
    const onChange = jest.fn()
    const wrapper = mount(createComponent({ onChange }))

    wrapper.find('input').simulate('change', { target: { value: '1.10' } })
    expect(onChange).toHaveBeenCalledTimes(1)
    expect(onChange).toBeCalledWith(1.10)
  })
})

function createComponent(props) {
  const defaultProps = {
    className: 'custom-class-name',
    name:      'field-name',
    onChange:  jest.fn(),
    value:     'value'
  }
  return <InputNumber {...defaultProps} {...props}/>
}
