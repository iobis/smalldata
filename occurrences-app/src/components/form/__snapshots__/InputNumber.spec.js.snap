// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText renders correctly 1`] = `
<InputNumber
  className="custom-class-name"
  name="field-name"
  onChange={[MockFunction]}
  value="value"
>
  <div
    className="column field custom-class-name"
  >
    <label
      className="label"
    >
      field-name.label
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
      value="value"
    />
  </div>
</InputNumber>
`;

exports[`InputText renders correctly if optional 1`] = `
<InputNumber
  className="custom-class-name"
  name="field-name"
  onChange={[MockFunction]}
  optional={true}
  value="value"
>
  <div
    className="column field custom-class-name"
  >
    <label
      className="label has-text-weight-normal"
    >
      field-name.label
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
      value="value"
    />
  </div>
</InputNumber>
`;
